default/python/AI/freeorion_tools/_freeorion_tools.py

```diff
@@ -347,3 +348,22 @@ def wrapper(*args, **kwargs):
 
         return wrapper
     return decorator
+
+
+def object_is_in_ion_storm(object_id):
+    # FIXME: This isn't working if we can not detect the ion storm
+    #        as is the case if the center of the storm is out of detection range
+    universe = fo.getUniverse()
+    universe_object = universe.getPlanet(object_id)
+    if not universe_object:
+        return False
+
+    for field_id in universe.fieldIDs:
```

Comment:
Before checking the fields, I would check if we own an object in the subject-object's system, and if so then return an early answer based entirely on whether that object is in a field (see if it has an ion field contribution to its stealth meter-- Effect Accounting for meters is directly exposed to python now although I can't recall having used it yet, and worst case we could do some reliable inference about our owned object).  Otherwise, we could make a wrong prediction (of either True or False) needlessly based on stale field object info.